### PR TITLE
Add docs constraints file

### PR DESCRIPTION
**Pull request by Czaki**
# Description
Add compilation of constraints files for docs build. Update workflow to use it. 
 

# References

## Type of change

- [x] Maintenance (changes required to run napari, tests, & CI smoothly)

### CODE REVIEW

File: .github/workflows/build_docs.yml [hunk 1]
```diff
@@ -60,7 +60,7 @@ jobs:
         with:
           # the napari-docs repo is cloned into a docs/ folder, hence the
           # invocation below. Locally, you should simply run make docs
-          run: make -C docs docs GALLERY_PATH=../examples/
+          run:  PIP_CONSTRAINTS=../napari-repo/resources/constraints/constraints_py3.10_docs.txt make -C docs docs GALLERY_PATH=../examples/
```
Comment (psobolewskiPhD) [inline .github/workflows/build_docs.yml]:
```suggestion
          run:  PIP_CONSTRAINTS=napari-repo/resources/constraints/constraints_py3.10_docs.txt make -C docs docs GALLERY_PATH=../examples/
```
I think that should fix the build?

Comment (Czaki) [reply .github/workflows/build_docs.yml]: It looks lie I wrongly understand how `make -C` works.

Comment (psobolewskiPhD) [reply .github/workflows/build_docs.yml]: The circleCI build where passing `PIP_CONSTRAINTS` isn't erroring, but isn't actually using it, see top lines of:
https://circleci.com/api/v1.1/project/github/napari/napari/786/output/106/0?file=true&allocation-id=64956571b63ae708434971e1-0-build%2F3FC75E41

```
echo clean
clean
echo /home/circleci/project/napari-docs/
/home/circleci/project/napari-docs/
rm -rf /home/circleci/project/napari-docs/docs/_build/
rm -rf /home/circleci/project/napari-docs/docs/api/napari*.rst
rm -rf /home/circleci/project/napari-docs/docs/gallery/*
rm -rf /home/circleci/project/napari-docs/docs/_tags
python -m pip install -qr /home/circleci/project/napari-docs/requirements.txt
python /home/circleci/project/napari-docs/docs/_scripts/prep_docs.py
Cloning into '/home/circleci/project/napari-docs/npe2'...
Warning: Permanently added the ECDSA host key for IP address '140.82.112.4' to the list of known hosts.
```

So I think we either need to change the Makefile or install using pip directly and then `make html` which skips the install.

Comment (psobolewskiPhD) [reply .github/workflows/build_docs.yml]: If the constraints file (the -docs one) is compiled with napari/docs requirements.txt and is used to install napari[dev] then do we need to use it again when installing the docs? everything should either be installed or compatible...

Comment (Czaki) [reply .github/workflows/build_docs.yml]: New release of docs dependencies may break env but the probability is lower.

Comment (psobolewskiPhD) [reply .github/workflows/build_docs.yml]: So maybe we should make a constraints in napari/docs also? and pip install using that and not requirements.txt?

File: .github/workflows/build_docs.yml [hunk 2]
```diff
@@ -57,10 +57,12 @@ jobs:
         env:
           GOOGLE_CALENDAR_ID: ${{ secrets.GOOGLE_CALENDAR_ID }}
           GOOGLE_CALENDAR_API_KEY: ${{ secrets.GOOGLE_CALENDAR_API_KEY }}
+          PIP_CONSTRAINTS: ${{ GITHUB_WORKSPACE }}/napari-repo/resources/constraints/constraints_py3.10_docs.txt
```
Comment (psobolewskiPhD) [inline .github/workflows/build_docs.yml]:
```suggestion
          PIP_CONSTRAINTS: ${{ env.GITHUB_WORKSPACE }}/napari-repo/resources/constraints/constraints_py3.10_docs.txt
```

Comment (psobolewskiPhD) [reply .github/workflows/build_docs.yml]: Not sure why it's not showing, but the build_docs workflow isn't valid:
https://github.com/napari/napari/actions/runs/5356030146

Comment (Czaki) [reply .github/workflows/build_docs.yml]: it looks working now

File: .github/workflows/build_docs.yml (hunk 1 above)
Comment (Czaki) [reply .github/workflows/build_docs.yml]: In each scenario, we use both repositories. I prefer to update constraints in only one of them.

**Review by psobolewskiPhD — APPROVED** [review]
We need to update the napari/docs workflow too right?
In this PR:
https://github.com/napari/docs/pull/213

**Comment by Czaki** [comment]
> We need to update the napari/docs workflow too right?

yes

**Pull request merged**